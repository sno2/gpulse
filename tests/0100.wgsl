// Infers result type
let fract_and_whole = modf(1.5);
// Sets fract_only to 0.5
let fract_only = modf(1.5).fract;
// Sets whole_only to 1.0
let whole_only = modf(1.5).whole;
