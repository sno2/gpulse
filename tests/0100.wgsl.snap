ast.Node{ .let_decl = ast.LetDecl{ .name = ast.Span{ .start = 26, .end = 41 }, .typ = null, .value = ast.Node{ .call = ast.Call{ .callee = ast.Node{ .identifier = ast.Span{ .start = 44, .end = 48 } }, .args = { ast.Node{ .number_literal = ast.Span{ .start = 49, .end = 52 } } } } } } } ast.Node{ .let_decl = ast.LetDecl{ .name = ast.Span{ .start = 85, .end = 95 }, .typ = null, .value = ast.Node{ .member = ast.BinaryOp{ .lhs = ast.Node{ .call = ast.Call{ .callee = ast.Node{ .identifier = ast.Span{ .start = 98, .end = 102 } }, .args = { ast.Node{ .number_literal = ast.Span{ .start = 103, .end = 106 } } } } }, .rhs = ast.Node{ .identifier = ast.Span{ .start = 108, .end = 113 } } } } } } ast.Node{ .let_decl = ast.LetDecl{ .name = ast.Span{ .start = 145, .end = 155 }, .typ = null, .value = ast.Node{ .member = ast.BinaryOp{ .lhs = ast.Node{ .call = ast.Call{ .callee = ast.Node{ .identifier = ast.Span{ .start = 158, .end = 162 } }, .args = { ast.Node{ .number_literal = ast.Span{ .start = 163, .end = 166 } } } } }, .rhs = ast.Node{ .identifier = ast.Span{ .start = 168, .end = 173 } } } } } } 